Added anchorPoint and rotation to Marker

yozef opened the pull request.

hansemannn commented on example/app.js:
```diff
@@ -10,7 +10,9 @@ var companies = {
         latitude: 37.368122,
-        longitude: -121.913653
+        longitude: -121.913653,
+		groundAnchor: {x:0.5, y:0.5},
```
Indentation of `groundAnchor` should be aligned with the other properties. Make sure to use spaces instead of tabs

hansemannn commented on iphone/Classes/TiGooglemapsMarkerProxy.m:
```diff
@@ -95,6 +95,23 @@ -(NSNumber*)flat
     return NUMBOOL([[self marker] isFlat]);
 }
 
+-(void)setGroundAnchor:(id)value
```
Use `args` instead of `value` to show that you provide multiple arguments.

hansemannn commented on iphone/Classes/TiGooglemapsMarkerProxy.m:
```diff
@@ -95,6 +95,23 @@ -(NSNumber*)flat
+
+-(NSNumber*)rotation
+{
+    return NUMBOOL([[self marker] rotation]);
```
Why does this return an boolean? The type of the property is a double. Use `NUMDOUBLE()` instead.

hansemannn commented on iphone/ti.googlemaps.xcodeproj/xcuserdata/theFool.xcuserdatad/xcschemes/ti.googlemaps.xcscheme:
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove this file please.

hansemannn commented on iphone/ti.googlemaps.xcodeproj/xcuserdata/theFool.xcuserdatad/xcschemes/xcschememanagement.plist:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Same as above, please remove.

hansemannn commented on iphone/ti.googlemaps.xcodeproj/xcuserdata/theFool.xcuserdatad/xcschemes/Build & Test.xcscheme:
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove this file please.

hansemannn commented on iphone/ti.googlemaps.xcodeproj/project.pbxproj:
```diff
@@ -564,7 +565,7 @@
 				GCC_WARN_UNUSED_VARIABLE = NO;
 				INSTALL_PATH = /usr/local/lib;
-				IPHONEOS_DEPLOYMENT_TARGET = 6.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
```
Is there a reason we need to change the deployment target?

hansemannn commented on iphone/manifest:
```diff
@@ -2,7 +2,7 @@
 # during compilation, packaging, distribution, etc.
 #
-version: 2.0.1
+version: 2.0.2
```
Consider using 2.1.0 instead of 2.0.2, since it's a new feature and not a bugfix.

hansemannn commented on iphone/module.xcconfig:
```diff
@@ -6,6 +6,6 @@
 // http://developer.apple.com/mac/library/documentation/DeveloperTools/Conceptual/XcodeBuildSystem/400-Build_Configurations/build_configs.html
 //
-FRAMEWORK_SEARCH_PATHS=$(SRCROOT)/../../modules/iphone/ti.googlemaps/2.0.1/platform/ios "~/Library/Application\ Support/Titanium/modules/iphone/ti.googlemaps/2.0.1/platform/ios"
+FRAMEWORK_SEARCH_PATHS=$(SRCROOT)/../../modules/iphone/ti.googlemaps/2.0.2/platform/ios "~/Library/Application\ Support/Titanium/modules/iphone/ti.googlemaps/2.0.2/platform/ios"
```
Same as above

hansemannn commented:
Hi @yozef, thank you for the PR! I left a few comments (mostly regarding unrelated files). To make PR's easier in the future, I will add a .gitignore now to handle user-specific files automatically. Thanks!

yozef commented:
hehehe I figured I committed way too many files for such small changes. My current module has more features implemented however, the latest GoogleMaps.bundle is over 100 mb. I'll figure out a way.

hansemannn commented:
Alright. The .gitignore was just pushed, so removing the builds in one commit should be future-save now. Any feature-requests are welcome!

yozef commented:
I'll be cleaning up more my code soon. I'm adding some of the animations (animateToLocation, animateToZoom, etc;). I'll be using this module heavily so might as well give back to the community and thanks to you we have a great head start :)

hansemannn commented:
Appreciate that. Will add the `infoWindowAnchor` property now and release the 2.1.0 after your changes can be merged.

yozef commented:
Great!

yozef commented:
Hi @hansemannn I've modified the files mentioned above. I've also added `bearing` and `viewingAngle` to `setCamera`, as well as the event trigger `mapViewDidFinishTileRendering` by listening to the `complete` event from Ti.

I will close this pull request in favor of the next one (which will be 2.1.0).

However before I commit, I wanted to add some methods of GoogleMaps, such as `animateToLocation`, and I think I'm doing something wrong, as it's not properly calling that method from Ti.
Here is what I've done:
In `TiGooglemapsMapViewProxy.m`, I added:
~~~
-(void) animateToLocation:(id)args {
    ENSURE_UI_THREAD_1_ARG(args);
    ENSURE_TYPE(args, NSDictionary);
    
    id latitude = [args valueForKey:@"latitude"];
    id longitude = [args valueForKey:@"longitude"];
    
    ENSURE_TYPE(latitude, NSNumber);
    ENSURE_TYPE(longitude, NSNumber);
    
    [[[self mapView] mapView] animateToLocation:CLLocationCoordinate2DMake([TiUtils doubleValue:latitude], [TiUtils doubleValue:longitude])];
}
~~~

and in `TiGooglemapsMapViewProxy.h`, I added:
`-(void)animateToLocation:(id)args;`

When I try to call from Ti this:
~~~
mapView.animateToLocation({
	latitude: 45.641959,
        longitude: -73.130588
});
~~~
Nothing happens (as in the Map dosent' animate to the specified longitude/latitude). I feel I'm missing something.

hansemannn commented:
@yozef I will take a look today.

yozef commented:
I'm starting to be convinced that referencing the `mapView` in `[[[self mapView] mapView] animateToLocation:CLLocationCoordinate2DMake([TiUtils doubleValue:latitude], [TiUtils doubleValue:longitude])];` is whats causing the issue.

hansemannn commented:
That looks fine. `[self mapView]` is the reference to the proxy view, the `[[self mapView] mapView]` references the `GMSMapView` inside the proxy view. Please try to use `ENSURE_UI_THREAD(animateToLocation, args)` to ensure it runs on the correct thread. This PR also has merge conflicts, so you might open another one to clean this up.

yozef commented:
I found the issue, I was missing a silly mistake `id location = [args objectAtIndex:0];`
Thanks!

for the new PR, the updated GoogleMaps framework is over 100 mb, and github is complaining that it's too large to commit. Maybe I could remove the framework from it's folder (and then other devs who wish to download it can do so and place it back in there)? Not sure if there are other solutions to by pass this limitation?
